approaches/abstract_approach.py

```diff
@@ -138,3 +167,21 @@ def get_duplicated_ids(self, query_num: int, topn: int) -> np.ndarray:
     @abstractmethod
     def update_history(self, query_num: int):
         raise NotImplementedError()
+
+    def __get_relevant_reports_num(self) -> Dict[str, int]:
+        result: Dict[str, int] = dict()
```

Comment:
You can use `collections.Counter` and avoid later checking key presence in the dict. If the stored data was something non-integer, you could've used `collections.defaultdict`

Reply:
Ok, now I use `collections.Counter`